### PR TITLE
Change some exception codes

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 289fcccbfc2cb456810361229881ddfcdad1efdd with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56316/289fcccbfc2cb456810361229881ddfcdad1efdd/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6749327448">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56316/289fcccbfc2cb456810361229881ddfcdad1efdd/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by vitlibar — APPROVED** [review]

File: src/Compression/ICompressionCodec.cpp [hunk 1]
```diff
@@ -125,7 +124,7 @@ UInt32 ICompressionCodec::readDecompressedBlockSize(const char * source)
 {
     UInt32 decompressed_block_size = unalignedLoadLittleEndian<UInt32>(&source[5]);
     if (decompressed_block_size == 0)
-        throw Exception(ErrorCodes::CORRUPTED_DATA, "Can't decompress data: header is corrupt with decompressed block size 0");
+        throw Exception(ErrorCodes::CANNOT_DECOMPRESS, "Can't decompress data: header is corrupt with decompressed block size 0");
```
Comment (vitlibar) [inline src/Compression/ICompressionCodec.cpp:127]: If this happens during unpacking stored parts of a table (e.g. during SELECT) it's still a very bad situation.

Comment (tavplubix) [reply src/Compression/ICompressionCodec.cpp:127]: Maybe we could pass a flag from `HTTPHandler` to `CompressedReadBuffer` and further, so codecs will throw `CANNOT_DECOMPRESS` for user's input and `CORRUPTED_DATA` for our internal data

**Review by alexey-milovidov — APPROVED** [review]

**Comment by tavplubix** [comment]
Integration tests (asan) [6/6] - https://github.com/ClickHouse/ClickHouse/issues/52554
Stress test (tsan) - https://github.com/ClickHouse/ClickHouse/pull/56388

**Pull request merged**